### PR TITLE
refactor: Remove `bluebird` from `lib/classes`

### PR DESCRIPTION
**Pull request by juanjoDiaz**
Addresses: #7171

### CODE REVIEW

**Comment by pgrzesik** [comment]
Hello @juanjoDiaz :wave: Thanks a lot for proposed changes. Before continuing with review, I have a question to @medikoo as I know you're currently working on refactoring variables resolution as a part of https://github.com/serverless/serverless/issues/8364 - is the refactor to `Variables.js` conflicting with your current work? Does it make sense to accept refactorings to that part at the moment or it's going to be removed soon anyway? Thanks :bow:

**Comment by medikoo** [comment]
> @medikoo as I know you're currently working on refactoring variables resolution as a part of #8364 - is the refactor to Variables.js conflicting with your current work?

@pgrzesik thanks for asking. I did not touch any part of old implementation. It's that new parser will simply take over, and older will only eventually serve as fallback, in case a new parser will report errors.

**Comment by juanjoDiaz** [comment]
Fixed the issues that @pgrzesik raised.

Any idea of why might the pipeline be failing?
My last commit only touched docs, and the build was passing before.

**Review by pgrzesik — COMMENTED** [review]
Hello @juanjoDiaz :wave: Thanks a lot - in general it looks good, but according to the original issue, in the first step we should address marking all promise-returning functions as `async` - here we have more refactoring, but not all promise-returning functions are marked as async. We also have a related PR that addreses that first part for `Variables.js` here: https://github.com/serverless/serverless/pull/8942 

To keep things in order, I believe the above PR should be resolved first and then we can continue with that refactoring. What is your opinion here @juanjoDiaz ?

**Comment by juanjoDiaz** [comment]
Oh I haven't seen that other PR.

I think that it makes sense to use `async` when you also use `await`.
I could review this PR and add async to any function that returns a promise but doesn't use async.

Apart from removing BbBluebird, this PR remove unnecessary promise wrappers and replace `then.catch` chains with `async/await`-style code.

If you rather waiting for the other PR and then check this, that's also fine.

File: lib/classes/Variables.js [hunk 1]
```diff
@@ -607,7 +604,7 @@ class Variables {
     if (requestedSlsVar === 'instanceId') {
       valueToPopulate = this.serverless.instanceId;
     }
-    return BbPromise.resolve(valueToPopulate);
+    return Promise.resolve(valueToPopulate);
```
Comment (pgrzesik) [inline lib/classes/Variables.js:607]: Can we just mark it as `async` and return the value?

File: lib/classes/Variables.js [hunk 2]
```diff
@@ -619,23 +616,23 @@ class Variables {
     } else {
       valueToPopulate = process.env;
     }
-    return BbPromise.resolve(valueToPopulate);
+    return Promise.resolve(valueToPopulate);
```
Comment (pgrzesik) [inline lib/classes/Variables.js:619]: Can we just mark it as `async` and return the value?

File: lib/classes/Variables.js [hunk 3]
```diff
@@ -619,23 +616,23 @@ class Variables {
...
   }
 
   getValueFromString(variableString) {
     // eslint-disable-line class-methods-use-this
     const valueToPopulate = variableString.replace(/^['"]|['"]$/g, '');
-    return BbPromise.resolve(valueToPopulate);
+    return Promise.resolve(valueToPopulate);
```
Comment (pgrzesik) [inline lib/classes/Variables.js:625]: Can we just mark it as `async` and return the value?

File: lib/classes/Variables.js [hunk 4]
```diff
@@ -619,23 +616,23 @@ class Variables {
...
   }
 
   getValueFromBool(variableString) {
     const valueToPopulate = variableString === 'true';
-    return BbPromise.resolve(valueToPopulate);
+    return Promise.resolve(valueToPopulate);
```
Comment (pgrzesik) [inline lib/classes/Variables.js:630]: Can we just mark it as `async` and return the value?

File: lib/classes/Variables.js [hunk 5]
```diff
@@ -619,23 +616,23 @@ class Variables {
...
   }
 
   getValueFromInt(variableString) {
     const valueToPopulate = parseInt(variableString, 10);
-    return BbPromise.resolve(valueToPopulate);
+    return Promise.resolve(valueToPopulate);
```
Comment (pgrzesik) [inline lib/classes/Variables.js:635]: Can we just mark it as `async` and return the value?

File: lib/classes/Variables.js [hunk 6]
```diff
@@ -646,7 +643,7 @@ class Variables {
     } else {
       valueToPopulate = this.options;
     }
-    return BbPromise.resolve(valueToPopulate);
+    return Promise.resolve(valueToPopulate);
```
Comment (pgrzesik) [inline lib/classes/Variables.js:646]: Can we just mark it as `async` and return the value?

File: lib/classes/Variables.js [hunk 7]
```diff
@@ -939,27 +927,24 @@ class Variables {
    * will later resolve to the deeper value
    */
   getDeeperValue(deepProperties, valueToPopulate) {
-    return BbPromise.reduce(
-      deepProperties,
-      (reducedValueParam, subProperty) => {
+    return Promise.resolve(
```
Comment (pgrzesik) [inline lib/classes/Variables.js:930]: Let's mark the function as async instead of using `Promise.resolve` construct

**Review by pgrzesik — CHANGES_REQUESTED** [review]
I agree that it makes sense - the problem with these changes is that they often can introduce bugs that are very hard to catch during review process, especially when the scope of the PR is bigger, that's why we suggested to migrate it gradually by first marking functions as `async`, removing unnecessary promise wrapers and only then getting rid of all BbPromise usages. But to not re-do the work you already did I think we should wrap up the changes as a part of this PR - the only thing that would be good here is if you could review the methods that are returning a promise and mark them as async if they're not marked as such yet. I also added a few comments related to `Promise.resolve` usage that I think we could get rid of

**Comment by juanjoDiaz** [comment]
I've marked all the functions returning promises as `async`.

I have the feeling that quite a few of the async functions in `Variables.js` are synchronous functions and the whole thing could be simplified but I didn't want to go there.

**Review by pgrzesik — CHANGES_REQUESTED** [review]
Thank you @juanjoDiaz - awesome job :raised_hands: However, I believe that I've made a mistake in my previous suggestions as I kinda missed the fact that these functions are not async at all - could you please check if we can just drop `async` from them? Other than that, we should be good to go :100: 

As for any further refactoring - at this point I think we shouldn't really refactor `Variables.js` as @medikoo is working on introducing totally new variables parser, which should make this one obsolete at some point in the future.

**Comment by juanjoDiaz** [comment]
Hi @pgrzesik,

The function as nor async but are called in a way that things break if you make them sync.

I was planning to raise a ticket about that once this PR is merged.
If the intention is to leave this as an obsolete class, I wouldn't really go into that change.

**Review by pgrzesik — APPROVED** [review]
Thank you @juanjoDiaz for verifying if we can drop `async` - all looks great now :bow:

**Pull request merged**